src/features/listing-builder/components/ListingBuilder/ListingBasic.tsx

```diff
@@ -80,17 +94,113 @@ export const ListingBasic = ({
   const [errorState, setErrorState] = useState<ErrorsBasic>({
     deadline: false,
     title: false,
+    slug: false,
     subSkills: false,
     skills: false,
     pocSocials: false,
     timeToComplete: false,
   });
 
+  const [isSlugGenerating, setIsSlugGenerating] = useState(false);
+  const [slugErrorMsg, setSlugErrorMsg] = useState('');
   const [isUrlValid, setIsUrlValid] = useState(true);
+  const [shouldSlugGenerate, setShouldSlugGenerate] = useState(false);
```

Comment:
there is no need for this
all you need to do is rerender only on title change by adding `title` as a dependancy to the useEffect

Reply:
no, its the flawed getUniqueSlug function. you are trying to rectify it by adding more code